Docs/developer_guide/mrml_overview.md

```diff
@@ -100,14 +100,346 @@ The following methods on the MRML scene are used to manage Undo/Redo stacks:
+* Use macro `vtkMRMLCopyContentMacro(vtkMRMLYourNodeType)` in the class definition and implement [`void CopyContent(vtkMRMLNode* anode, bool deepCopy)`](http://apidocs.slicer.org/master/classvtkMRMLNode.html#a94940453f557fbc3a59f65bd37311cef) in the cxx file. This should copy the data content of your node via either a shallow or deep copy.
+* Override [`vtkMRMLStorageNode* CreateDefaultStorageNode()`](http://apidocs.slicer.org/master/classvtkMRMLStorableNode.html#aa35937e99fbb721847c3aea7a818fa10) to return an owning pointer default storage node type for your class (see [The storage node](#the-storage-node)).
+* Override [`void CreateDefaultDisplayNodes()`](http://apidocs.slicer.org/master/classvtkMRMLDisplayableNode.html#ac9f1e28f999d5ccc035317246b728b03) to create the default display nodes (for 3D and/or 2D viewing).
+* Override [`void ProcessMRMLEvents(vtkObject * caller, unsigned long event, void* callData)`](http://apidocs.slicer.org/master/classvtkMRMLNode.html#afce3fc71df4f8d1e56901846f3c86a51):
+  * This is used to process any events that happen regarding this object.
+  * This method should handle the `vtkMRMLTransformableNode::TransformModifiedEvent` which is emitted any time the transform that is associated with the data object is changed.
+* Convenience methods - while not necessarily needed, they are nice to have.
+  * `Get<MyCustomType>DisplayNode()` function that returns the downcast version of [`GetDisplayNode()`](http://apidocs.slicer.org/master/classvtkMRMLDisplayableNode.html#ab2cac46d934a10817b75b37471fe65a8) saves users of your class a bit of downcasting.
+* Other methods:
+  * Add other methods as your heart desires to view/modify the actual content of the data being stored.
+
+:::{tip}
+
+Any methods with signatures that contain only primitives, raw pointers to VTK derived objects, or a few std library items like `std::vector` will be automatically wrapped for use in Python. Any functions signatures that contain other classes (custom classes, smart pointers from the std library, etc) will not be wrapped. For best results, try to use existing VTK data objects, or have your custom classes derive from [`vtkObject`](https://vtk.org/doc/nightly/html/classvtkObject.html) to get automatic wrapping.
+
+:::
+
+#### The display node
+
+The display node, contrary to what one may think, is not actually how a MRML object is displayed on screen. Instead it is the list of options for displaying a MRML object. Things like colors, visibility, opacity; all of these can be found in the display node.
+
+Files:
+
+* \<Extension>/\<Module>/MRML/vtkMRML\<MyCustomType>DisplayNode.h
+* \<Extension>/\<Module>/MRML/vtkMRML\<MyCustomType>DisplayNode.cxx
```

Comment:
```
|-- <Extension>
       |-- <Module>
              |-- MRML
                    |-- vtkMRML<MyCustomType>DisplayNode.h
                    |-- vtkMRML<MyCustomType>DisplayNode.cxx
```

Reply:
Updated to suggested file structure formatting.